rdr_service/tools/tool_libs/genomic_utils.py

```diff
@@ -493,6 +493,86 @@ def run(self):
         return 0
 
 
+class ManualSampleClass(GenomicManifestBase):
+    """
+    Class for inserting arbitrary genomic samples manually. Used for E2E testing.
+    """
+    def __init__(self, args, gcp_env: GCPEnvConfigObject):
+        super(ManualSampleClass, self).__init__(args, gcp_env)
+
+    def run(self):
+        """
+        Main program process
+        :return: Exit code value
+        """
+
+        # Validate Aruguments
+        if self.args.csv is None:
+            _logger.error('Argument --csv must be provided.')
+            return 1
+
+        if not os.path.exists(self.args.csv):
+            _logger.error(f'File {self.args.csv} was not found.')
+            return 1
+
+        # Activate the SQL Proxy
+        self.gcp_env.activate_sql_proxy()
+        self.dao = GenomicSetMemberDao()
+
+        # Update gsm IDs from file
+        with open(self.args.csv, encoding='utf-8-sig') as f:
+            lines = f.readlines()
+
+            if not self.args.dryrun:
+                new_genomic_set = GenomicSet(
+                    genomicSetName=f"New Manual Sample List-{self.nowf}",
+                    genomicSetCriteria=".",
+                    genomicSetVersion=1,
+                )
+
+                with self.dao.session() as session:
+                    inserted_set = session.merge(new_genomic_set)
+                    session.flush()
+
+                    for line in lines:
+                        if len(line.split(',')) > 0:
+                            _pid = line.split(',')[0].strip()
+                            _bid = line.split(',')[1].strip()
+                            _sample_id = line.split(',')[2].strip()
+                            _sab = line.split(',')[3].strip()
+                            _siteId = line.split(',')[4].strip()
+                            _state = line.split(',')[5].strip()
```

Comment:
Would a csvreader work for parsing the file?

Reply:
I thought that was the name of some class, but it looks like it doesn't really exist. `csv.reader` https://docs.python.org/3/library/csv.html is what I meant by it.

Reply:
Done.